From 5ac9dd024f27dfecf13f17336c82e912df35db0e Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Tue, 27 Jan 2026 05:53:24 +0530
Subject: [PATCH 8/8] giomodule: Do not try to use GIO_MODULE_DIR on iOS

When we try to load that dir on a real phone, nothing happens because
the path doesn't exist since the app is a statically-linked bundle.
When we try on the simulator, we get warnings when the compiled-in
path exists in the user machine because the modules fail to load.
---
 gio/giomodule.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gio/giomodule.c b/gio/giomodule.c
index 782d633..7edcbae 100644
--- a/gio/giomodule.c
+++ b/gio/giomodule.c
@@ -1281,8 +1281,10 @@ get_gio_module_dir_darwin (void)
           return g_steal_pointer (&possible_dir);
         }
     }
-#endif
   return g_strdup (GIO_MODULE_DIR);
+#else
+  return NULL;
+#endif
 }
 #endif
 
-- 
2.52.0

